### PR TITLE
nixos/freshrss: fix permissions and add database test

### LINKED ISSUE #207728: freshrss installation is not working
### Describe the bug
Hi first sorry for not using the template the correct way, i hope its okay.
At the moment i try to install an freshrss instance on my system.
i have an first installed 22.05 and updated to 22.11 nixos system.
beside of there is no manual how to install it and what are pre requirements i have to need, here is my configuration file:
```
{ config, pkgs, ... }:

{
  services.freshrss = {
    enable = true;
    package = pkgs.freshrss;
    language = "en";
    defaultUser = "picard";
    passwordFile = "/run/secrets/freshrss";
    dataDir = "/var/lib/freshrss";
    baseUrl = "https://freshrss.localhost";
    virtualHost = "freshrss";
    database = {
      user = "spock";
      port = 3306;
      name = "freshrss";
      host = "localhost";
      tableprefix = "freshrss";
#      passFile = "/run/secrets/freshrss";
      type = "pgsql";
    };
  };
  services.postgresql = {
    enable = true;
    package = pkgs.postgresql;
    dataDir = "/mnt/postgresql1/";
    ensureDatabases = [ "freshrss" ];
    ensureUsers = [
      {
        name = "freshrss";
        ensurePermissions = {
          "DATABASE freshrss" = "ALL PRIVILEGES";
        };
      }
      {
        name = "superuser";
        ensurePermissions = {
          "ALL TABLES IN SCHEMA public" = "ALL PRIVILEGES";
        };
      }
    ];
    port = 3306;
  };
}
```

I have the following problem at the moment:
When i use the default dataDir directory (the one whos standing in the config atm) than freshrss can create the needed files and directorys but the systemd fails with the following error:
freshrss_setup_fail
and i dont come further anymore.

But when i use an complete other dataDir like "/mnt/freshrss/" (what i want to do) than i get the following systemd error:
freshrss_setup_fail2
the service is still not working (with the same error) even when i create the /mnt/freshrss/ folder and even when i give the folder 777 permission and give it the freshrss user and group.
freshrss_setup_fail3

### Notify maintainers
@Stunkymonkey 

### Metadata
```console
[root@NCC-1031:/mnt]# nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.83, NixOS, 22.11 (Raccoon), 22.11.1105.9898811c658`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"home-manager-22.05.tar.gz, nixos-22.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Description of changes

fixes #207728

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/freshrss.nix [hunk 1]
```diff
@@ -116,12 +116,18 @@ in
...
+      default = "freshrss";
+      description = lib.mdDoc "User under which Freshrss runs.";
+    };
+  };
 
   config =
     let
-      systemd-hardening = {
+      defaultServiceConfig = {
+        ReadWritePaths = "$(cfg.dataDir)";
```
Comment (miangraham) [inline nixos/modules/services/web-apps/freshrss.nix:130]:
```suggestion
        ReadWritePaths = "${cfg.dataDir}";
```


As is:

```
Jan 06 21:54:29 futaba systemd[1]: /etc/systemd/system/freshrss-updater.service:33: ReadWritePaths= path is not absolute, ignoring: $(cfg.dataDir)
Jan 06 21:54:29 futaba systemd[1]: /etc/systemd/system/freshrss-config.service:32: ReadWritePaths= path is not absolute, ignoring: $(cfg.dataDir)

$ grep ReadWritePaths /etc/systemd/system/freshrss-updater.service
ReadWritePaths=$(cfg.dataDir)
```

Comment (miangraham) [reply nixos/modules/services/web-apps/freshrss.nix:130]: Also more of a side note, but even with the red log text above things seem to be working (tested with sqlite). You may not need this bit at all.

**Comment by miangraham** [comment]
Result of `nixpkgs-review pr 209254` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>

File: nixos/modules/services/web-apps/freshrss.nix (hunk 1 above)
Comment (Stunkymonkey) [reply nixos/modules/services/web-apps/freshrss.nix:130]: thanks. this is needed so we still have `ProtectSystem` set to strict.

**Comment by miangraham** [comment]
0e27878 works too, ReadWritePaths confirmed fixed!

**Review by miangraham — APPROVED** [review]

**Pull request merged**